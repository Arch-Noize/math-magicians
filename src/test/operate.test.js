import operate from '../logic/operate';

describe('Test the operates', () => {
  test('you should subtract the numbers', () => {
    const a = 4;
    const b = 2;
    const operation = '-';

    expect(operate(a, b, operation)).toMatchSnapshot();
  });

  test('you should sum the numbers', () => {
    const a = 1;
    const b = 2;
    const operation = '+';

    expect(operate(a, b, operation)).toMatchSnapshot();
  });

  test('you should divide the numbers', () => {
    const a = 6;
    const b = 2;
    const operation = 'รท';

    expect(operate(a, b, operation)).toMatchSnapshot();
  });

  test('you should calculate the modulus of the numbers', () => {
    const a = 2;
    const b = 1;
    const operation = '%';

    expect(operate(a, b, operation)).toMatchSnapshot();
  });

  test('you should multiply the numbers', () => {
    const a = 2;
    const b = 1;
    const operation = 'x';

    expect(operate(a, b, operation)).toMatchSnapshot();
  });
});
