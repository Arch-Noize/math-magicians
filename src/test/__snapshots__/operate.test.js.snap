// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the operates you should calculate the modulus of the numbers 1`] = `"0"`;

exports[`Test the operates you should divide the numbers 1`] = `"3"`;

exports[`Test the operates you should multiply the numbers 1`] = `"2"`;

exports[`Test the operates you should subtract the numbers 1`] = `"2"`;

exports[`Test the operates you should sum the numbers 1`] = `"3"`;
